File: Lecture 2/bmi/app/src/main/java/com/example/bmi/WelcomeScreen.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

public class WelcomeScreen extends AppCompatActivity {
    public static int SPLASH_TIME_OUT = 2000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome_screen);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run(){
                Intent homeIntent = new Intent(WelcomeScreen.this, MainActivity.class);
                startActivity(homeIntent);
                finish();
            }
        },SPLASH_TIME_OUT);
    }
}

File: Lecture 4/bmi/app/src/main/java/com/example/bmi/WelcomeScreen.kt
package com.example.bmi

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity

class WelcomeScreen : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome_screen)
        Handler().postDelayed(object : Runnable() {
            override fun run() {
                val homeIntent = Intent(this@WelcomeScreen, MainActivity::class.java)
                startActivity(homeIntent)
                finish()
            }
        }, SPLASH_TIME_OUT)
    }

    companion object {
        var SPLASH_TIME_OUT = 2000
    }
}